docs/rules/function-name-format.md

```diff
@@ -0,0 +1,122 @@
+# Function Name Format
+
+Rule `function-name-format` will enforce a convention for function names.
+
+## Options
+
+* `allow-leading-underscore`: `true`/`false` (defaults to `true`)
+* `convention`: `'hyphenatedlowercase'` (default), `camelcase`, `snakecase`, or a Regular Expression that the function name must match (e.g. `^[_A-Z]+$`)
+* `convention-explanation`: Custom explanation to display to the user if a function doesn't adhere to the convention
+
+## Example 1
+
+Settings:
+- `allow-leading-underscore: true`
+- `convention: hyphenatedlowercase`
+
+When enabled, the following are allowed:
+
+```scss
+$foo: hyphenated-lowercase();
```

Comment:
As with the tests I would prefer it if we used function declarations in the documentation as that is what most people will be familiar with and expect to see in the documentation. The current examples are just assigning the output of a function to variable and may lead to confusion.

```scss
@function do-something($test) {
  @return $test;
}
```